application/features/bootstrap/UserContext.php

```diff
@@ -72,18 +72,22 @@ private function buildClient(): Client
     private function sendRequest(Client $client, string $action, string $resource): ResponseInterface
     {
         switch ($action) {
-            case 'created': return $client->post  ($resource);
-            case 'updated': return $client->put   ($resource);
-            case 'deleted': return $client->delete($resource);
-                   default: return $client->get   ($resource);
+            case 'created'  : return $client->post  ($resource);
+            case 'updated'  : return $client->put   ($resource);
+            case 'deleted'  : return $client->delete($resource);
+            case 'retrieved': return $client->get   ($resource);
+            case 'headed'   : return $client->head  ($resource);
+            case 'patched'  : return $client->patch ($resource);
+
+            default: throw new InvalidArgumentException("$action is not a recognized HTTP verb.");
         }
     }
 
     private function extractBody(Response $response): array
     {
         $jsonBlob = $response->getBody()->getContents();
 
-        return json_decode($jsonBlob, true);
+        return json_decode($jsonBlob, true) ?? [];
```

Comment:
It looks like this will silently ignore failures to decode the JSON string. Is that the intent?